File: app/src/main/java/com/example/rickmortymvvm/PresentationCharacterListAction.java
package com.example.rickmortymvvm;

public abstract class PresentationCharacterListAction {
     public static class GoToInfo extends PresentationCharacterListAction {
          private final Character character;

          public GoToInfo(Character character) {
               this.character = character;
          }

          public Character getCharacter() {
               return character;
          }
     }

     public static class Finish extends PresentationCharacterListAction {

     }
}

File: app/src/main/java/com/example/rickmortymvvm/presentation/PresentationCharacterListAction.kt
package com.example.rickmortymvvm.presentation

import com.example.rickmortymvvm.presentation.models.CharacterVM

sealed class PresentationCharacterListAction {
    data class GoToInfo(val characterVM: CharacterVM) : PresentationCharacterListAction()
    object Finish : PresentationCharacterListAction()
}
